cmd/oras/internal/option/packer.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/pflag"
+	"oras.land/oras-go/v2/content"
+)
+
+const (
+	annotationManifest = "$manifest"
+)
+
+var (
+	errAnnotationConflict    = errors.New("`--annotation` and `--annotation-file` cannot be both specified")
+	errAnnotationFormat      = errors.New("missing key in `--annotation` flag")
+	errAnnotationDuplication = errors.New("duplicate annotation key")
+)
+
+// Packer option struct.
+type Packer struct {
+	ManifestExportPath     string
+	PathValidationDisabled bool
+	AnnotationFilePath     string
+	ManifestAnnotations    []string
+
+	FileRefs []string
+}
+
+// ApplyFlags applies flags to a command flag set.
```

Comment:
Just a clarifying question here! In the first implementation I have, the user could provide an annotations file, and this wasn't just for the manifest, you could have keys `$manifest` or `$config` or the fullpath/relative path of a blob to indicate that we want to add annotations there. But now the new argument `--annotation key=value` seems to be scoped just for the final manifest, and not a config or particular blob. Is this a correct interpretation?

Reply:
And in this case, wouldn't we only care if both are defined given we have `$manifest` in the file lookup plus an `--annotation`? E.g., providing this new flag with an annotation file without manifest annotations should not be an issue, but as I see here the client is just disallowing both to be defined.

Reply:
Hello @vsoch thank you for insight.

1. If users want to update annotations including keys `$manifest`, `$config` and more, they can use `--annotation-file` and provide an annotations file
2. If users only want to update manifest annotations, they can use `--annotation`. They don't have to create a file just to upload a single annotation
3. I have set up a condition that `--annotation` and `--annotation-file` cannot be both specified to make sure their implements are isolated to each other

Reply:
Yes so my question is why not to allow a file with $manifest and a one off annotation that has a key that doesn’t overlap? It could be a use case given that someone has a file with common/shared annotations and then a few custom to a container.  I think this should be allowed because there is no complexity in terms of duplicate keys - it’s just reading a file and then adding one off annotations from the command line. And you’d immediately reject/exit given duplicates only.